nemo/collections/asr/models/clustering_diarizer.py

```diff
@@ -259,10 +259,18 @@ def _run_vad(self, manifest_file):
             num_workers=self._cfg.num_workers,
         )
         AUDIO_VAD_RTTM_MAP = deepcopy(self.AUDIO_RTTM_MAP.copy())
-        for key in AUDIO_VAD_RTTM_MAP:
-            AUDIO_VAD_RTTM_MAP[key]['rttm_filepath'] = os.path.join(table_out_dir, key + ".txt")
+        input_uniq_id_list = []
+        for uniq_id in AUDIO_VAD_RTTM_MAP:
+            AUDIO_VAD_RTTM_MAP[uniq_id]['rttm_filepath'] = os.path.join(table_out_dir, uniq_id + ".txt")
+            input_uniq_id_list.append(uniq_id)
 
         write_rttm2manifest(AUDIO_VAD_RTTM_MAP, self._vad_out_file)
+
+        # Error handling for empty audio samples.
```

Comment:
Why to handle this condition twice? You are deleting uniq_ids in `write_rttm2manifest` and again after that as well. Can't we gather all silence uniq_ids and delete from AUDIO_RTTM_MAP at once?